pages/tools/index.js

```diff
@@ -4,33 +4,56 @@ import GenericLayout from '../../components/layout/GenericLayout';
 import Heading from '../../components/typography/Heading';
 import Paragraph from '../../components/typography/Paragraph';
 import TextLink from '../../components/typography/TextLink';
+// import loader from '../public/img/loaders/log-primary.png';
+import { useEffect, useState } from 'react';
 
 export default function toolsDashboard() {
   const description = 'Tools Dashboard for AsyncAPI Initiative';
   const image = '/img/social/meetings.png';
-  
+  const loader = 'img/loaders/logo-primary.png';
+  // const loader='spinning-circle.svg';
+
+  const [loading, setLoading] = useState(false);
+  useEffect(() => {
+    setLoading(true);
+    setTimeout(() => {
+      setLoading(false);
+    }, 1000);
+  }, []);
+
   return (
     <div>
-      <GenericLayout title="Tools" description={description} image={image}>
```

Comment:
Preloader should be a part of `GenericLayout`, not outside the layout of the page so that Navbar is shown while data is loading in the page.